Python/Product/Analysis/LanguageServer/Server.cs

```diff
@@ -480,9 +483,7 @@ private sealed class ReloadModulesQueueItem : IAnalyzable {
                 return Task.FromResult(actualItem);
             }
 
-            if (_clientCaps?.python?.analysisUpdates ?? false) {
-                pyItem.OnNewAnalysis += ProjectEntry_OnNewAnalysis;
-            }
+            pyItem.OnNewAnalysis += (o, e) => OnPythonEntryNewAnalysis(pyItem);
```

Comment:
Why change the event handler signature? We don't need the lambda and closure here, and for large workspaces it's a cost we may notice.

Reply:
Since `OnNewAnalysis` is a public API, I can't change its signature and provide correct type in `eventArgs`. Type check inside old handler was misleading - there can be no other type, so this code is cleaner and easier to understand (no need to look for all existing implementations of `IPythonProjectTree`). Closure contains two reference type fields, so I can't imagine how we may notice it considering hundreds of kilobytes of garbage strings in G2 per every new tree parse.